Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -10,6 +10,13 @@ class Cask
     def initialize(token, sourcefile_path: nil, &block)
       @token = token
       @sourcefile_path = sourcefile_path
+
+      @tap = if @sourcefile_path == CaskLoader.default_path(@token)
+        Hbc.default_tap
+      elsif @sourcefile_path.to_s =~ HOMEBREW_TAP_PATH_REGEX
+        Tap.fetch(Regexp.last_match(1), Regexp.last_match(2))
+      end
```

Comment:
Sure thing—just pushed that change. It's a little funky to me now that the cask_spec is also testing `CaskLoader` logic, but it was doing that before so I think it's fine.